SDK/Android/BugfenderBinding.cs

```diff
@@ -1,188 +1,182 @@
+        {
+            Com.Bugfender.Sdk.Bugfender.SetDeviceInteger(key, new Java.Lang.Integer(value));
+        }
+
+        public void SetDeviceFloat(string key, float value)
+        {
+            Com.Bugfender.Sdk.Bugfender.SetDeviceFloat(key, new Java.Lang.Float(value));
+        }
+
+        public void RemoveDeviceKey(string key)
+        {
+            Com.Bugfender.Sdk.Bugfender.RemoveDeviceKey(key);
+        }
+
+        public void Log(int lineNumber, string method, string file, LogLevel logLevel, string tag, string message)
+        {
+            // a negative lineNumber indicates we need to guess from the stack
+            if (lineNumber < 0)
+            {
+                System.Diagnostics.StackFrame frame = new System.Diagnostics.StackTrace(true).GetFrame(2);
+                if (frame.GetType().Namespace.StartsWith("Com.Bugfender"))
+                {
+                    frame = new System.Diagnostics.StackTrace(true).GetFrame(3);
+                }
+
+                lineNumber = frame.GetFileLineNumber();
+                method = frame.GetMethod().Name;
+                file = System.IO.Path.GetFileName(frame.GetFileName());
+                if (method == null) method = "";
+                if (file == null) file = "";
+            }
+
+            Com.Bugfender.Sdk.Bugfender.Log(lineNumber, method, file, MapLoglLevelToSdkLogLevel(logLevel), tag, message);
+        }
+
+        private static Com.Bugfender.Sdk.LogLevel MapLoglLevelToSdkLogLevel(LogLevel logLevel)
+        {
+            switch (logLevel)
+            {
+                case LogLevel.Fatal:
+                    return Com.Bugfender.Sdk.LogLevel.Fatal;
+                case LogLevel.Error:
+                    return Com.Bugfender.Sdk.LogLevel.Error;
+                case LogLevel.Warning:
+                    return Com.Bugfender.Sdk.LogLevel.Warning;
+                case LogLevel.Info:
+                    return Com.Bugfender.Sdk.LogLevel.Info;
+                case LogLevel.Debug:
+                    return Com.Bugfender.Sdk.LogLevel.Debug;
+                case LogLevel.Trace:
+                    return Com.Bugfender.Sdk.LogLevel.Trace;
+                default:
+                    return Com.Bugfender.Sdk.LogLevel.Info;
+            }
+        }
+
+        public void ForceSendOnce()
+        {
+            Com.Bugfender.Sdk.Bugfender.ForceSendOnce();
+        }
+
+        public Uri SendIssue(string title, string markdown)
+        {
+            Java.Net.URL url = Com.Bugfender.Sdk.Bugfender.SendIssue(title, markdown);
+            if (url == null)
+                return null;
+            return new Uri(url.ToString());
+        }
+
+        public Uri SendCrash(string title, string text)
+        {
+            Java.Net.URL url = Com.Bugfender.Sdk.Bugfender.SendCrash(title, text);
+            if (url == null)
+                return null;
+            return new Uri(url.ToString());
+        }
+
+        /* TODO

```

Comment:
Should this be here?

Reply:
This TODO hasn't changed (already present in previous version). I don't think it's wrong for it to be here until someone takes the time to address it.